Fix duplicate posting of permit url

0xcodercrane opened the pull request.

pavlovcik commented on src/handlers/comment/handlers/payout.ts:
```diff
@@ -27,7 +27,7 @@ export const payout = async (body: string) => {
 
   const IssueComments = await getAllIssueComments(issue.number);
-  if (IssueComments === null) {
+  if (IssueComments.length === 0) {
```
Should it check for `null` and empty array? Why was it `null` before?

0xcodercrane replied:
I am not sure why it was `null` someone made it. but `getAllIssueComments` returns an array of comments so no need to make the return object `null`,  empty array is better to handle.

pavlovcik replied:
According to git blame it was @Steveantor. 

Can you explain why you put `null` here?

pavlovcik replied:
```suggestion
  if (IssueComments?.length === 0) {
```

This also solves for both `null` and empty array.

0xcodercrane replied:
https://github.com/ubiquity/ubiquibot/pull/344#issuecomment-1567720041

pavlovcik commented on src/handlers/payout/index.ts:
```diff
@@ -72,6 +72,13 @@ export const handleIssueClosed = async () => {
+  const exist = commentContents.find((content) => content.includes(comment));
+  if (exist) {
+    logger.info(`Skip to generate a permit url because it has been already posted`);
+    return `Skip to generate a permit url because it has been already posted`;
```
```suggestion
    return `Permit generation skipped because it was already posted to this issue.`;
```

pavlovcik commented on src/helpers/issue.ts:
```diff
@@ -99,11 +99,11 @@ export const getCommentsOfIssue = async (issue_number: number): Promise<Comment[
 };
 
-export const getAllIssueComments = async (issue_number: number): Promise<Comment[] | null> => {
+export const getAllIssueComments = async (issue_number: number): Promise<Comment[]> => {
   const context = getBotContext();
   const payload = context.payload as Payload;
 
-  let result: Comment[] | null = [];
```
These also seem obviously intentional. I believe that the API sometimes returns `null` @0xcodercrane

0xcodercrane replied:
Its getting handle properly here.

0xcodercrane replied:
https://github.com/ubiquity/ubiquibot/pull/344#issuecomment-1567720041

0xcodercrane commented:
Why did I remove `null` from the following codebase? 
`export const getAllIssueComments = async (issue_number: number): Promise<Comment[] | null> => {}`

There's no mandatory requests in code style so its free to make it as you want. but I removed `null` from the following benefits.

1/ Simplify the function body itself.
```typescript
- export const getAllIssueComments = async (issue_number: number): Promise<Comment[] | null> => {
+ export const getAllIssueComments = async (issue_number: number): Promise<Comment[]> => {

- let result: Comment[] | null = [];
+ let result: Comment[] = [];

catch (e: unknown) {
-   result = unknown
    shouldFetch = false;
}
```

2/ make its usage simpler
I've added both lines to use it. 
```
  const comments = await getAllIssueComments(issue.number);
  const commentContents = comments.map((i) => i.body);
  const exist = commentContents.find((content) => content.includes(comment));
```
If `getAllIssueComments` still returns `null`, I had to add a null check for both lines.
